### PR TITLE
[Core] Fix ScalingSolver considers always a scale for symmetric matrix 

### LINKED ISSUE #3327: [Core] ScalingSolver considers always a scale for symmetric matrix
https://github.com/KratosMultiphysics/Kratos/blob/acfffbbb30aa44bf7984df4e03498499c21ffe73/kratos/linear_solvers/scaling_solver.h#L105

Independently of the the option given to the constructor it always considers the option as true

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #3327

### CODE REVIEW

File: kratos/linear_solvers/scaling_solver.h [hunk 1]
```diff
@@ -56,7 +56,6 @@ namespace Kratos
  * @tparam TSparseSpaceType The sparse space definition
  * @tparam TDenseSpaceType The dense space definition
  * @tparam TReordererType The reorder considered
- * @warning SymmetricScaling is not taken as argument. Check that
```
Comment (loumalouomega) [inline kratos/linear_solvers/scaling_solver.h:59]: BTW I already noticed when implementing the factory, I just remember now

**Comment by loumalouomega** [comment]
Anyone?

**Review by philbucher — APPROVED** [review]
this makes sense to me, thanks

**Review by RiccardoRossi — APPROVED** [review]

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**